packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +189,176 @@ class Tooltip extends mixin(
+              focusinEventName,
+              this._handleFocusin,
+              !hasFocusin
+            )
+          );
+        } else {
+          this.element.focus();
+        }
+
+        if (typeof callback === 'function') callback();
+      }
     );
   }
 
   /**
-   * Attaches event handlers to show/hide the tooltip.
+   * Attaches event handlers to show the tooltip.
    * @param {Element} element The element to attach the events to.
    * @private
    */
   _hookOn(element) {
-    const hasFocusin = 'onfocusin' in window;
-    const focusinEventName = hasFocusin ? 'focusin' : 'focus';
-    [focusinEventName, 'blur', 'touchleave', 'touchcancel'].forEach(name => {
+    /**
+     * Setup the _debouncedHandleClick function for displaying a tooltip
+     * @param {Event} evt - user initiated event
+     * @param {Integer[]} [allowedKeys] - allowed key codes the user may press to open the tooltip
+     * @private
+     */
+    const setupDebounceClick = (evt, allowedKeys) => {
+      const { relatedTarget, type, which } = evt;
+      // Allow user to use `space` or `enter` to open tooltip
+      if (typeof allowedKeys === 'undefined' || allowedKeys.includes(which)) {
+        const hadContextMenu = this._hasContextMenu;
+        this._hasContextMenu = type === 'contextmenu';
+        this._debouncedHandleClick({
+          relatedTarget,
+          type,
+          hadContextMenu,
+          details: getLaunchingDetails(evt),
+        });
+      }
+    };
+
+    this.manage(on(element, 'click', setupDebounceClick, false));
+
+    if (this.element.tagName !== 'BUTTON') {
       this.manage(
         on(
-          element,
-          name,
+          this.element,
+          // Does Carbon prefer keydown or keyup?
+          'keydown',
           event => {
-            const { relatedTarget, type } = event;
-            const hadContextMenu = this._hasContextMenu;
-            this._hasContextMenu = type === 'contextmenu';
-            this._debouncedHandleClick({
-              relatedTarget,
-              type: type === 'focusin' ? 'focus' : type,
-              hadContextMenu,
-              details: getLaunchingDetails(event),
-            });
+            setupDebounceClick(event, this._allowedOpenKeys);
           },
-          name === focusinEventName && !hasFocusin
+          false
         )
       );
-    });
+    }
+  }
+
+  /**
+   * Attaches event handlers to hide the tooltip.
+   * @param {Element} element The element to attach the events to.
+   * @private
+   */
+  _hookCloseActions(element) {
+    this._handleClickListener = this.manage(
+      on(element.ownerDocument, 'click', evt => {
+        if (
+          !eventMatches(
+            evt,
+            element.getAttribute(this.options.attribTooltipTarget)
+          )
+        ) {
+          this.changeState('hidden', evt);
+        }
+      })
+    );
+
+    this._handleKeydownListener = this.manage(
+      on(element.ownerDocument, 'keydown', evt => {
```

Comment:
Probably we can implement ESC key handler in `floating-menu.js` code given overflow menu would need ESC key handler, too?